File: pset1/mario/more/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void) 

{

    // prompt user for height of pyramid
    int height;
    do
    {

        height = get_int("Height: ");

    } 
    while (height <= 0 || height > 8);

    // create a row with every iteration 
    int line = 1;
    for (int rows = 0; rows < height; rows++)
    {
        // for each row
        // print spaces
        int spaces = height - line;
        for (int i = 0; i < spaces; i++) 
        {
            printf(" ");
        } 

        // print left hashes
        for (int j = 0; j < line; j++) 
        {   
            printf("#");
        } 

        // print gap
        for (int gap = 0; gap < 2; gap++) 
        {   
            printf(" ");
        } 

        // print right hashes
        for (int r = 0; r < line; r++) 
        {   
            printf("#");
        } 
      
        printf("\n");
        line++;
    }

}

File: pset6/mario/more/mario.py
from cs50 import get_int

# Prompt for height until requirements are met
while True:
    height = get_int("Height: ")
    if height > 0 and height < 9:
        break

# Iterate over rows
for row in range(height):

    # Left pyramid
    for left_space in range((height - 1) - row):
        print(" ", end="")

    for left_hash in range(row + 1):
        print("#", end="")

    # Gap
    print("  ", end="")

    # Right pyramid
    for right_hash in range(row + 1):
        print("#", end="")

    hashes = height - row
    for right_space in range(hashes):
        print("", end="")

    # New line
    print()